function/aggregate/aggregate.go

```diff
@@ -64,10 +88,11 @@ func addToGroup(rows []group, series api.Timeseries, tags []string) []group {
 }
 
 // groupBy groups the given SeriesList by tags, producing a list of lists (of type groupResult)
-func groupBy(list api.SeriesList, tags []string) []group {
+// if `combines` is true, then it groups on all other tags instead.
+func groupBy(list api.SeriesList, tags []string, combines bool) []group {
 	result := []group{}
 	for _, series := range list.Series {
-		result = addToGroup(result, series, tags)
+		result = addToGroup(result, series, tags, combines)
```

Comment:
both `addToGroup`'s new tag generation and `groupAccepts` are starting to become unwieldy. we're using the boolean `combines` to do function dispatching at this point.

there should be a better way to organize these function. i feel like addToGroup should accept two functions (one to project one tag set to another grouped tagset), and don't pass in `tags` or `combines` at all. - if we have that, `groupAccept` can use that projection function to compare the tags also, simplifying `if combine...` sprinkled around everywhere.